### PR TITLE
feat: add deploy list

### PR DESCRIPTION
**Pull request by sokel**
Smart contract with array of allowed addresses, which may deploy contracts.

### CODE REVIEW

File: blockchain/source/contracts/DeployList.sol [hunk 1]
```diff
@@ -0,0 +1,37 @@
+pragma solidity ^0.4.23;
+
+
+import "zeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+contract DeployList is Ownable {
+    address[]deployers;
```
Comment (sshaman1101) [inline blockchain/source/contracts/DeployList.sol]: the `addresses` is an array of `deployers` or the `deployers` is an array of `addresses`?

File: blockchain/source/contracts/DeployList.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        DeployerAdded(_deployer);
+    }
+
+    function removeDeployer(address _deployer) public onlyOwner {
+        for (uint i = 0; i < deployers.length - 1; i++)
+            if (deployers[i] == _deployer) {
+                deployers[i] = deployers[deployers.length - 1];
+                break;
+            }
+        deployers.length -= 1;
```
Comment (sshaman1101) [inline blockchain/source/contracts/DeployList.sol]: It completely weird to change array length by hand, why not to use a mapping here?

File: blockchain/source/contracts/DeployList.sol [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        DeployerRemoved(_deployer);
```
Comment (sshaman1101) [inline blockchain/source/contracts/DeployList.sol]: emit

File: blockchain/source/contracts/DeployList.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    event DeployerRemoved(address deployer);
+
+    constructor(address[] _deployers){
+        owner = msg.sender;
+        deployers = _deployers;
+    }
+
+    function addDeployer(address _deployer) public onlyOwner {
+        deployers.push(_deployer);
+        DeployerAdded(_deployer);
```
Comment (sshaman1101) [inline blockchain/source/contracts/DeployList.sol]: emit

File: blockchain/source/test/deployers.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    let deployer3 = accounts[4];
+
+    let newDeployer = accounts[6];
+
+    before(async function () {
+        list = await DeployList.new([deployer1, deployer2, deployer3], { from: owner });
+    });
+
+    describe('Default', function () {
+        it('', async function () {
```
Comment (sshaman1101) [inline blockchain/source/test/deployers.js]: it should what?

File: blockchain/source/test/deployers.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            assert.equal(addresses[1], deployer2);
+            assert.equal(addresses[2], deployer3);
+        });
+    });
+
+    describe('AddDeployer', function () {
+        it('should add new element', async function () {
+            await list.addDeployer(newDeployer);
+            let addresses = await list.getDeployers();
+            assert.notEqual(addresses.indexOf(newDeployer), -1);
```
Comment (sshaman1101) [inline blockchain/source/test/deployers.js]: does your test suite have smth like `assert.Contains` or `assert.In`

File: blockchain/source/test/deployers.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        });
+    });
+
+    describe('RemoveDeployer', function () {
+        it('should remove element', async function () {
+            await list.removeDeployer(deployer1);
+            let addresses = await list.getDeployers();
+            assert.equal(addresses.indexOf(deployer1), -1);
```
Comment (sshaman1101) [inline blockchain/source/test/deployers.js]: same as above

File: blockchain/source/contracts/DeployList.sol (hunk 2 above)
Comment (sokel) [reply blockchain/source/contracts/DeployList.sol]: fuuuuuuuuuuuuuuuuu

**Pull request merged**